### ISSUE TITLE
Help on checking the `functionCall_3.sol` test case

### ISSUE THREAD

**Issue by shangweilin** [kframework/solidity-semantics#31]
In the `tests/functionCall_3.sol` test case (in the `NewConfig-SimpleExp-Function` branch), consider the statement 
```
int sum = functionCall(new C();f;2, 1);
```
The `functionCall`, which is an expression, is not evaluated in the current rules.

Compared with the `tests/functionCall_2.sol` test case, the following statement works
```
int sum; sum = functionCall(new C();f;2, 1);
```

**Comment by dfilaretti** [comment]
@shangweilin
I think the problem is due to variable _initialisation_. 

Although the two examples looks similar, they actually do two different things. 

In the second example the variable is first initialised to some initial/default value (`int sum;`) and then it is assigned a different value (`sum = functionCall(new C();f;2, 1);`): 

```
int sum;                                               // initialise to undef 
sum = functionCall(new C();f;2, 1);      // assign new value functionCall(new C();f;2, 1)
```

In the first example, we only have a variable initialisation, where it turns out the initial value in this case is a complex expression like `functionCall(new C();f;2, 1)`:

```
int sum = functionCall(new C();f;2, 1); // initialise to functionCall(new C();f;2, 1)
```

If I understand correctly, the rule for initialisation that is invoked here is:

```
rule
    <k> T:TypeName S:StorageLocations X:Id = E => #allocate(C, CN, #varInfo(X:Id, E,T:TypeName, #mem)) ~> #end_Exp ...</k>
    <currentAccount> #account(C:Int, CN:Id) </currentAccount>
```
(notice that `E` here is `functionCall(new C();f;2, 1)`) 

Now, `#allocate` gets stuck because it expects a `Value`, but `functionCall` is not one:

```
rule [Allocate-Local-Variables]:
    <k> #allocate(C:Int, CN:Id, #varInfo(X:Id, E:Value, T:TypeName, #mem)) => . ...</k>
    <env> ENV:Map => ENV[X <- #storedVar(!N:Int, T, #mem)] </env>
    <currentAccount> #account(C, CN) </currentAccount>
    <localMem> MEM:Map => MEM[!N <- E] </localMem>

```

So basically you need to add something so that that expression gets evaluated. 
You can try using strictness so that `E` is fully evaluated, so eventually you get a simple `Value`.

**Comment by shangweilin** [comment]
Hi, @dfilaretti,
Many thanks for your hint. I have fixed this issue.
It turns out that the strictness of `VariableDefinition` is missing. The issue is solved if the syntax of `VariableDefinition` is modified as follows:

```
syntax VariableDefinition ::= "var" Id
                            | VariableDeclaration
                            | VariableDeclaration "=" Expression    [strict(2)]
```

I guess we can close this issue now.
